### PR TITLE
Set computed value of sideEffects in package.json for modules

### PR DESCRIPTION
**Pull request by GoodDayForSurf**

### CODE REVIEW

File: packages/devextreme/build/gulp/transpile.js [hunk 1]
```diff
@@ -104,6 +105,15 @@ const createModuleConfig = (name, dir, filePath) => {
     return JSON.stringify(result, null, 2);
 };
 
+function hasSideEffects(jsFileModulePath) {
+    try {
+        const data = fs.readFileSync(jsFileModulePath, 'utf8');
+
+        return !(/^\s*export /ms).test(data)
+    } catch (e) {
+        return true;
```
Comment (ivanblinov2k17) [inline packages/devextreme/build/gulp/transpile.js]: Maybe log error message and then return true?

**Review by dxvladislavvolkov — DISMISSED** [review]

File: packages/devextreme/build/gulp/side-effects-finder.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+class SideEffectFinder {
+    #checkingPathsStack = new Set();
+    #checkedPaths = new Map();
+
+    getModuleSideEffectFiles(moduleFilePath) {
```
Comment (dxvladislavvolkov) [inline packages/devextreme/build/gulp/side-effects-finder.js:10]: Maybe rename the func to getSideEffectModules or getSideEffectFiles

File: packages/devextreme/build/gulp/side-effects-finder.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        try {
+            const moduleSideEffectFiles = [...this.#findSideEffectsInModule(moduleFilePath)]
```
Comment (dxvladislavvolkov) [inline packages/devextreme/build/gulp/side-effects-finder.js]: Also rename

File: packages/devextreme/build/gulp/side-effects-finder.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+class SideEffectFinder {
+    #checkingPathsStack = new Set();
```
Comment (ivanblinov2k17) [inline packages/devextreme/build/gulp/side-effects-finder.js]: remove variable to smth like recursionPathStack

**Review by ivanblinov2k17 — APPROVED** [review]

**Review by dxvladislavvolkov — APPROVED** [review]

**Pull request merged**